appserver/jdbc/jdbc-config/src/main/java/org/glassfish/jdbc/config/validators/JdbcConnectionPoolValidator.java

```diff
@@ -65,23 +68,75 @@ public void initialize(final JdbcConnectionPoolConstraint constraint) {
         this.poolFaults = constraint.value();
     }
 
+    public String getParsedVariable(String variableToRetrieve) {
+
+        String[] variableReference = variableToRetrieve.split("=");
+        
+        switch(variableReference[0]) {
+            case "ENV":
+                //Check environment variables for requested value
+                if (System.getenv(variableReference[1]) != null && !System.getenv(variableReference[1]).isEmpty()) {
+                    return System.getenv(variableReference[1]);
+                }
+                break;
+            case "MPCONFIG":
+                //Check microprofile config for requested value
+                Config config = ConfigProvider.getConfig();
+                if (config.getValue(variableReference[1], String.class) != null && !config.getValue(variableReference[1], String.class).isEmpty()) {
```

Comment:
`config.getValue(variableReference[1], String.class)` should be extracted to local var